src/machine/machine_atsamd21.go

```diff
@@ -136,6 +136,10 @@ var pinPadMapping = [32]byte{
 // found" (indicated by returning ok=false). The pad number is returned to
 // calculate the DOPO/DIPO bitfields of the various serial peripherals.
 func findPinPadMapping(sercom uint8, pin Pin) (pinMode PinMode, pad uint32, ok bool) {
+	if len(pinPadMapping) <= int(pin)/2 {
+		return
```

Comment:
I would suggest:

1. Adding a comment explaining why this early return is necessary.
2. (style nitpick) swapping the condition, this is the order that I more commonly see.

```suggestion
	if int(pin)/2 >= len(pinPadMapping) {
		// This is probably NoPin, for which no mapping is available.
		return
```